PowerEditor/src/Parameters.cpp

```diff
@@ -651,38 +651,23 @@ typedef void (WINAPI *PGNSI)(LPSYSTEM_INFO);
 
 void cutString(const TCHAR* str2cut, vector<generic_string>& patternVect)
 {
-	TCHAR str2scan[MAX_PATH];
-	lstrcpy(str2scan, str2cut);
-	size_t len = lstrlen(str2scan);
-	bool isProcessing = false;
-	TCHAR *pBegin = nullptr;
+	const TCHAR *pBegin = str2cut;
+	const TCHAR *pEnd = pBegin;
 
-	for (size_t i = 0 ; i <= len ; ++i)
+	while (*pEnd != '\0')
 	{
-		switch(str2scan[i])
+		if (*pEnd == ' ')
 		{
-			case ' ':
-			case '\0':
-			{
-				if (isProcessing)
-				{
-					str2scan[i] = '\0';
-					patternVect.push_back(pBegin);
-					isProcessing = false;
-				}
-				break;
-			}
-
-			default:
-			{
-				if (!isProcessing)
-				{
-					isProcessing = true;
-					pBegin = str2scan+i;
-				}
-			}
+			if (pBegin != pEnd)
+				patternVect.emplace_back(pBegin, pEnd);
+			pBegin = pEnd + 1;

```

Comment:
Seems this version is not reading over input separating the text with multiple spaces, which seems to be the case in the original version.

Reply:
It will work fine, have tested this case - if words are separated with multiple spaces, pBegin keeps being decremented together with pEnd and the inequality comparision will keep failing. Thus even if separated with more than one space, words are tokenized fine.

I have also verified that output is valid if the string begins or ends with spaces.

Reply:
Thanks for the countercheck. I will close my PR in favor of this one.

Reply:
What do you think about replacing
`if (*pEnd == ' ')`
with
`if (*pEnd <= ' ')`
?

This way tokenizing would work on all whitespaces, not just space - which may be preferable for those who edit UDL manually.

Reply:
In this case maybe usage of _istspace() , see https://msdn.microsoft.com/en-US/library/y13z34da.aspx, would do it more reliably.

Reply:
Good idea! I'll update the code accordingly to use it.